README.md

```diff
@@ -131,46 +127,40 @@ Now that we have our `getUserData` function, let's import it to our component wh
 
 ### Lifecycle and rendering
 
-We've heard about React lifecycle methods already, they are a place to run functions related to the components lifecycle. In our case we'll want to run a data fetching function as the component mounts the dom. Sounds familiar? Let's define `componentDidMount()` where we'll call `getUserData`.
+We've heard about React effects already, they are a place to do side-effects such as fetching data. In our case we'll want to run a data fetching function once the component mounts into the dom. Sounds familiar? Let's create an effect with `React.useEffect()` where we'll call `getUserData`.
 
-```javascript=
-componentDidMount() {
+```javascript
+React.useEffect(() => {
   const username = "sofiapoh";
   getUserData(username).then(data => console.log(data));
-}
+}, []);
 ```
 
-Hopefully you're seeing something in your console by now! That alone is not enough for us to get rendering, we need to set this data in our components `data` state variable we defined earlier in order for us to consume it outside `componentDidMount`.
+Don't forget the second argument to `useEffect`—the empty array. This will tell React to only run the effect once (you don't want to make thousands of requests in a row and get rate limited by Github).
```

Comment:
```suggestion
Don't forget the second argument to `useEffect`—the empty array. This will tell React to only run the effect once. Without this your component will re-render every time you update your state, which will re-run the effect, which will update state, which will re-render your component, which will re-run the effect... (you don't want to make thousands of requests in a row and get rate limited by Github).
```